COMET/misc_plugins/PlotScripts/analysis_scripts/IV_PQC.py

```diff
@@ -1,82 +1,671 @@
+
+                    # See if the r2 value has increased and store end points
+                    if r2_left >= LR2:
+                        LR2 = r2_left
+                        LeftEndPoints = ((df["xaxis"][0], intercept_left),(df["xaxis"][idx], slope_left * df["xaxis"][idx] + intercept_left))
+
+        # Find the right fit by averaging on the final 20 points
+        average_right = np.mean(list(df['yaxis'][-20:]))
+        RightEndPoints =[(df['xaxis'][len(df['xaxis'])-20],average_right),(df['xaxis'][len(df['xaxis'])-1],average_right)]
+
+        # Find the line intersection
+        full_depletion_voltages = line_intersection(LeftEndPoints, RightEndPoints)
+        fdepvoltage.append(full_depletion_voltages[0])
+        self.log.info('Full depletion voltage: {} V'.format(np.round(full_depletion_voltages[0], 2)))
+
+        # Add vertical line for full depletion
+        vline = hv.VLine(full_depletion_voltages[0]).opts(color='black', line_width=5.0)
+
+        # Add slopes
+        left_line = np.array([[0, np.median(Left_stats[:,2])],[full_depletion_voltages[0], full_depletion_voltages[1]]])
+        left_line = hv.Curve(left_line).opts(color='grey')
+        right_line = hv.HLine(average_right).opts(color='grey')
+
+        # Add text
+        text = hv.Text(230, (5e+21), 'Depletion voltage: {} V'.format(np.round(full_depletion_voltages[0], 2))).opts(style=dict(text_font_size='20pt'))
+
+        # Update the plot specific options if need be
+        returnPlot = plot * vline * right_line * left_line * text
+        returnPlot = customize_plot(returnPlot, "1C2", configs["IV_PQC"], **addConfigs)
+
+        return returnPlot, full_depletion_voltages[0]
+
+    def plot(self, diodePlots, mos_Plots, gate_Plots, cv_files, diode_files, gate_files):
+        # Select the plots to show depending on the kind of files you have
+        if len(cv_files) != 0 and len(diode_files) != 0 and len(gate_files) != 0:
+            self.PlotDict['All'] = gate_Plots + diodePlots + mos_Plots
+        elif len(cv_files) != 0 and len(diode_files) != 0 and len(gate_files) == 0:
+            self.PlotDict['All'] = diodePlots + mos_Plots
+        elif len(cv_files) == 0 and len(diode_files) != 0 and len(gate_files) == 0:
+            self.PlotDict['All'] = diodePlots
+        elif len(cv_files) != 0 and len(diode_files) == 0 and len(gate_files) == 0:
+            self.PlotDict['All'] = mos_Plots
+        elif len(cv_files) == 0 and len(diode_files) == 0 and len(gate_files) != 0:
+            self.PlotDict['All'] = gate_Plots
+        elif len(cv_files) != 0 and len(diode_files) == 0 and len(gate_files) != 0:
+            self.PlotDict['All'] = gate_Plots + mos_Plots
+        else:
+            self.PlotDict['All'] = diodePlots + gate_Plots
+        self.PlotDict["All"] = applyPlotOptions(self.PlotDict["All"], {'Curve': {'color': "hv.Palette('Category20')"}})  # To change colors
+
+        return self.PlotDict["All"]
+
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/dominic.bloech/UniDAQ/pullRequest?prid=5481807)